api/operator/v1beta1/vmextra_types.go

```diff
@@ -1069,6 +1069,11 @@ type CommonDefaultableParams struct {
 	// Has priority over `VM_DISABLESELFSERVICESCRAPECREATION` operator env variable
 	// +optional
 	DisableSelfServiceScrape *bool `json:"disableSelfServiceScrape,omitempty"`
+
+	// Use endpointslices in discoveryrole instead of endpoints
+	// for the vmservicescrape.
+	// +optional
+	VMServiceScrapeDefaultRoleEndpointslices *bool `json:"vmServiceScrapeDefaultRoleEndpointslices,omitempty"`
```

Comment:
`CommonDefaultableParams` is a common struct for multiple CRs like vmcluster..., so we shouldn't add non-common field here.